presto-main/src/main/java/com/facebook/presto/execution/ClusterSizeMonitor.java

```diff
@@ -131,20 +165,20 @@ public synchronized void verifyInitialMinimumWorkersRequirement()
      */
     public synchronized ListenableFuture<?> waitForMinimumWorkers()
     {
-        if (currentCount >= executionMinCount) {
+        if (currentWorkerCount >= workerMinCount) {
             return immediateFuture(null);
         }
 
         SettableFuture<?> future = SettableFuture.create();
-        futures.add(future);
+        workerSizeFutures.add(future);
 
         // if future does not finish in wait period, complete with an exception
         ScheduledFuture<?> timeoutTask = executor.schedule(
                 () -> {
                     synchronized (this) {
                         future.setException(new PrestoException(
                                 GENERIC_INSUFFICIENT_RESOURCES,
-                                format("Insufficient active worker nodes. Waited %s for at least %s workers, but only %s workers are active", executionMaxWait, executionMinCount, currentCount)));
+                                format("Insufficient active worker nodes. Waited %s for at least %s workers, but only %s workers are active", executionMaxWait, workerMinCount, currentWorkerCount)));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: currentWorkerCount, workerMinCount. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 181. Consider formatting this data with &#39;%d&#39; instead.